iocore/net/UnixNetVConnection.cc

```diff
@@ -80,7 +80,14 @@ read_signal_and_update(int event, UnixNetVConnection *vc)
 {
   vc->recursion++;
   if (vc->read.vio.cont) {
-    vc->read.vio.cont->handleEvent(event, &vc->read.vio);
+    // Most of the time the continuation will have the same mutex as the vc, but not always
```

Comment:
I understand the necessity but I don't think this magic shouldn't be spread out here and there. Are there cases that `vc->read.vio.cont` have a different mutex with any valid reasons? If not, the root cause should be fixed.

It would be nice if we can generalize these 7 lines and use it at other similar places.

Reply:
This kind of logic is used in a lot of places, I think the generalization that would be most useful would be one that automatically reschedules if the lock on a Continuation can not be obtained. There was some investigation on providing lock queues on Continuations and/or proxy mutexes for performance reasons but was dropped because only the cache really has significant lock contention. Perhaps we should look at it again from a code cleanliness point of view.

Reply:
Agreed this is ugly and in my testing it is causing events of unexpected types showing up in different places.  I think the concrete issue in this case, is that the origin connection is being reused, so it's mutex is not the same as the user-agent and state machine's mutex.